pymodbus/client/base.py

```diff
@@ -183,6 +197,10 @@ async def async_execute(self, request) -> ModbusResponse:
 
         return resp  # type: ignore[return-value]
 
+    async def async_execute(self, request) -> ModbusResponse:
+        """Execute requests asynchronously with framer."""
+        return await self.framer.asyncProcess(self._async_execute(request))
```

Comment:
Seems you are mixing architectural layers ! framer is responsible to build the frame not to execute.

The problem is that as we discussed earlier, the queue is in rtu_framer, but architectural it belongs in the transaction layer, which already have a queue concept in place (even though it depends on the device to handle the actual queue).